p2p/host/match.go

```diff
@@ -0,0 +1,35 @@
+package host
+
+import (
+	"strings"
+
+	semver "github.com/coreos/go-semver/semver"
+)
+
+func MultistreamSemverMatcher(base string) (func(string) bool, error) {
+	parts := strings.Split(base, "/")
+	vers, err := semver.NewVersion(parts[len(parts)-1])
+	if err != nil {
+		return nil, err
+	}
+
+	return func(check string) bool {
+		chparts := strings.Split(check, "/")
+		if len(chparts) != len(parts) {
+			return false
+		}
+
+		for i, v := range chparts[:len(chparts)-1] {
+			if parts[i] != v {
+				return false
+			}
+		}
+
+		chvers, err := semver.NewVersion(chparts[len(chparts)-1])
+		if err != nil {
+			return false
+		}
+
+		return vers.Major == chvers.Major && vers.Minor >= chvers.Minor
```

Comment:
Ah, i remember now. If you dial to `1.1.0` as `1.0.0`, the other side should reply with `1.0.0`, allowing you to continue on as normal